File: app/dto/product-info.dto.ts
import type { BaseDTO } from "./base.dto";
import type { CustomerSubscriptionDTO } from "./customer-subscription.dto";

export interface ProductInfoDTO extends BaseDTO {
    id?: number;
    storeId?: string;
    storeName?: string;
    productHandle?: string;
    productId?: number;
    productTitle?: string;
    variantId?: number;
    variantTitle?: string;
    price?: number;
    imageURL?: string;
    status?: boolean;
    inStock?: boolean;
    customerSubscribe: CustomerSubscriptionDTO;
    vendor?: string;
    createdAt?: Date;
    updatedAt?: Date;
    isActive?: boolean;
}

File: finally-bis-cron/src/main/java/com/nine15/fbis/dto/ProductInfoDTO.java
package com.nine15.fbis.dto;

import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

@Data
@Setter
@Getter
@Builder
public class ProductInfoDTO {
    private String storeId;
    private String storeName;
    private String productHandle;
    private Long productId;
    private String productTitle;
    private Long variantId;
    private String variantTitle;
    private Double price;
    private String imageURL;
    private Boolean status;
    private Boolean inStock;
    private Boolean isActive;

}
